File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
    if (start <= 0) {
        // Вернуть ошибку или исключение
        return -1; // или какое-то другое значение, обозначающее ошибку
    }
    
    int res = 0;
    while (start != 1) {
        if (start % 2 == 1) {
            start = 3 * start + 1; 
        } else {
            start /= 2;
        }
        ++res;
    }

    return res;
}

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "fmt"

func CollatzConjecture(n int) (int, error) {
	if n <= 0 {
        return -1, fmt.Errorf("incorrect value %d", n)
    }
    res := 0
    for n != 1 {
        if n % 2 == 0 {
            n /= 2
        } else {
            n = 3 * n + 1
        }
        res++
    }

    return res, nil
}
